checked_message_bench_test.go

```diff
@@ -0,0 +1,96 @@
+package zap_test
+
+import (
+	"testing"
+
+	"github.com/uber-go/atomic"
+	"github.com/uber-go/zap"
+)
+
+func BenchmarkCheckedMessage_Chain(b *testing.B) {
+	infoLog := zap.New(
+		zap.NullEncoder(),
+		zap.InfoLevel,
+		zap.DiscardOutput,
+	)
+	errorLog := zap.New(
+		zap.NullEncoder(),
+		zap.ErrorLevel,
+		zap.DiscardOutput,
+	)
+
+	data := []struct {
+		lvl zap.Level
+		msg string
+	}{
+		{zap.DebugLevel, "meh"},
+		{zap.InfoLevel, "fwiw"},
+		{zap.ErrorLevel, "hey!"},
+	}
+
+	p := atomic.NewInt32(0)
+
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		i := 0
+		j := p.Inc()
+		myInfoLog := infoLog.With(zap.Int("p", int(j)))
+		myErrorLog := errorLog.With(zap.Int("p", int(j)))
+		for pb.Next() {
+			d := data[i%len(data)]
+			cm := myInfoLog.Check(d.lvl, d.msg)
+			cm = cm.Chain(myErrorLog.Check(d.lvl, d.msg))
+			if cm.OK() {
+				cm.Write(zap.Int("i", i))
+			}
+			i++
+		}
+	})
+}
+
+func BenchmarkCheckedMessage_Chain_sliceLoggers(b *testing.B) {
+	logs := []zap.Logger{
+		zap.New(
+			zap.NullEncoder(),
+			zap.InfoLevel,
+			zap.DiscardOutput,
+		),
+		zap.New(
```

Comment:
should we move the common logic into a private function?
```go
newBenchmarkLogger(options ...zap.Option)
```

It would pass the `NullEncoder()` and `DiscardOutput` by default